src/guest/guest-session.service.ts

```
@@ -157,6 +166,41 @@ export class GuestSessionService implements OnModuleInit {
+   * caller must surface 503, not a misleading 401/500. On the in-memory
+   * fallback (`redisStore` undefined) this is always false: that store cannot
+   * have a connectivity outage, so misses there are real.
+   */
+  private isStorageUnavailable(): boolean {
+    // `.client` is typed as a client|cluster union; only the standalone client
+    // exposes `isReady`. Read it structurally so either shape compiles, and
+    // treat "explicitly not ready" as the only outage signal (undefined ->
+    // don't know -> not an outage).
+    const client = this.redisStore?.client as
+      | { isReady?: boolean }
+      | undefined;
```

ESLint

[error] Replace `⏎······|·{·isReady?:·boolean·}⏎·····` with `·{·isReady?:·boolean·}`

(prettier/prettier)